JCLOUDS-813: fix list private images

andreaturli opened the pull request:
To fix the bug reported at JCLOUDS-813 this PR revisit the strategy to listImages:
- modify the listImages and getImage strategies in SoftLayerComputeServiceAdapter: by default, it only consider `stock` images
- add support for VirtualGuestBlockDeviceTemplateGroupApi getObject: this is need in `getImage` if the user specify a specific `imageId` to scan public and private images + stock
- remove PROPERTY_SO FTLAYER_INCLUDE_PUBLIC_IMAGES: this is no more needed

andreaturli commented:
@nacx can you have another look at it? thanks!

nacx commented on providers/softlayer/src/main/java/org/jclouds/softlayer/features/VirtualGuestBlockDeviceTemplateGroupApi.java:
```diff
@@ -51,8 +52,17 @@
+   @GET
+   @Path("/SoftLayer_Virtual_Guest_Block_Device_Template_Group/{id}/getObject")
+   @QueryParams(keys = "objectMask", values = MASK)
+   VirtualGuestBlockDeviceTemplateGroup getObject(@PathParam("id") String id);
```
Add the corresponding unit and live tests for this new api method.

nacx commented on providers/softlayer/src/main/java/org/jclouds/softlayer/features/VirtualGuestBlockDeviceTemplateGroupApi.java:
```diff
@@ -51,8 +52,17 @@
    Set<VirtualGuestBlockDeviceTemplateGroup> getPublicImages();
 
+   /**
+    * @return public images
```
This returns a single image.

andreaturli commented:
@nacx I've addressed your comments, sorry I forgot about the tests.

I need to port the expectTests to MWS but not in this PR

andreaturli commented:
@nacx if the builder is happy can backport this to 1.8.x as well?

nacx commented:
Feel free to backport it to 1.8.x, but it is unlikely that another 1.8 version is released.

andreaturli commented:
Great, @nacx! The builder is happy, can I merge it now?

andreaturli commented:
Great, @nacx! Squashed and rebased but the builder is not happy, because of an unrelated issue. Closing and re-opening the PR to trigger the builder again

nacx commented:
+1 once the build is green again. Thx @andreaturli !
